cloudinit/config/cc_ssh.py

```diff
@@ -43,7 +43,6 @@
 
 Supported public key types for the ``ssh_authorized_keys`` are:
 
-    - dsa
```

Comment:
@holmanb do you think it's worth us documenting here that DSA host key generation dropped in 23.4, or shall we prefer downstreams to rely on the [upstream project Changelog (and release changelog) here](https://github.com/canonical/cloud-init/blob/main/ChangeLog)

Reply:
I lean towards the later. If a user does actually exist, and upgrading cloud-init breaks their configuration, then looking at the docs will tell them that no such key exists. Given the unlikelyhood of this still being used I doubt it is worth it, and as you pointed out - the changelogs will document this change, as will the commit history and the change in docs between versions.